geomesa-filter/src/main/scala/org/locationtech/geomesa/filter/FilterHelper.scala

```diff
@@ -217,6 +217,10 @@ object FilterHelper extends LazyLogging {
           val buffered = filter match {
             // note: the dwithin should have already between rewritten
             case dwithin: DWithin => geom.buffer(dwithin.getDistance)
+            case bbox: BBOX =>
+              val geomCopy = gf.createGeometry(geom)
```

Comment:
yeah, addWayPointsToBBOX below does modify the geometry.  these lines are basically taken from method above doing similar: visitBinarySpatialOp